### ISSUE TITLE
[Research] LSTEEG Deep Autoencoder for Real-Time EEG Artifact Rejection (Replace Threshold-Based)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#34]
## Paper
**Title:** EEG Artifact Detection and Correction with Deep Autoencoders (LSTEEG)
**Authors:** Starlab team
**Year:** 2025
**Venue:** arXiv (2502.08686)
**URL:** https://arxiv.org/abs/2502.08686

## Also Relevant
**Title:** EEGEpochNet: self-supervised contrastive learning for automated EEG epoch rejection
**Year:** 2025
**Venue:** PubMed (recent)
**URL:** https://pubmed.ncbi.nlm.nih.gov/41730244/

## Key Contribution
**LSTEEG** is a novel LSTM-based autoencoder for multi-channel EEG artifact detection and correction. It treats artifact detection as anomaly detection — trained only on clean EEG, then flags deviations as artifacts. Outperforms convolutional autoencoders in both detection and correction.

**EEGEpochNet** achieves F1 scores of 93-95% on automated epoch rejection using self-supervised contrastive learning — no labeled artifact data needed. Three modules: multi-branch 1D-CNN (morphological features), bidirectional GRU (temporal), and contrastive loss for domain-invariant representations.

## Relevance to NeuralDreamWorkshop
Current artifact handling is threshold-based (75 uV amplitude, frozen EMA on bad epoch). This misses subtle artifacts (slow electrode drift, low-amplitude EMG) and rejects too aggressively on high-amplitude clean signals. A learned artifact model would:
1. Detect artifacts that threshold rules miss (especially subtle EMG at AF7/AF8)
2. Correct rather than reject — reconstruct clean signal from noisy input
3. Adapt to individual users' EEG characteristics (self-supervised)

### Files to Create
- `ml/models/lsteeg_artifact.py` — LSTM autoencoder for artifact detection + correction
  - Encoder: 2-layer bidirectional LSTM (hidden=64) → latent space (dim=32)
  - Decoder: 2-layer LSTM → reconstructed clean EEG
  - Anomaly score: reconstruction error (MSE) — high error = artifact
- `ml/training/train_lsteeg.py` — Train on clean EEG segments from ISRUC/Sleep-EDF

### Files to Modify
- `ml/processing/eeg_processor.py`:
  - Add `lsteeg_clean(signal, model)` function alongside existing `preprocess()`
  - Integrate into pipeline: raw → bandpass → notch → **LSTEEG correction** → feature extraction
  - Keep threshold-based as fallback when model not loaded
- `ml/models/artifact_classifier.py` — Add LSTEEG anomaly score as additional feature for supervised classifier
- `ml/api/routes.py`:
  - Modify `/denoise` endpoint to support LSTEEG mode
  - Add `/artifacts/auto-detect` endpoint using anomaly scoring
  - Return `artifact_probability` and `corrected_signal` in response
- `ml/models/denoising_autoencoder.py` — Compare LSTEEG vs existing conv autoencoder, keep best

### Architecture
```
Input: Multi-channel EEG segment (4 x 256 samples = 1 second)
  → Bidirectional LSTM Encoder (2 layers, hidden=64)
  → Latent representation (dim=32)
  → LSTM Decoder (2 layers)
  → Reconstructed clean EEG (4 x 256)

Artifact Detection: MSE(input, reconstruction) > threshold → artifact
Artifact Correction: Use reconstruction as cleaned signal
```

### Self-Supervised Training (EEGEpochNet approach)
- Collect clean resting-state EEG from user's first 2-min calibration
- Train personal artifact model on user's own clean EEG (few-shot)
- No labeled artifact data needed — just clean examples

## Expected Impact
- **Artifact detection F1:** 85-93% (up from ~70% with threshold rules)
- **Signal quality:** 20 dB SNR improvement on corrected signals (per LSTEEG paper)
- **Emotion accuracy:** +5-10% by feeding cleaner signal to downstream models
- **Key win:** Corrects artifacts rather than rejecting entire epochs (no data loss)

## Complexity: MEDIUM
- LSTM autoencoder is a standard PyTorch architecture (~150 lines)
- Training on clean EEG is straightforward (no labels needed)
- Integration into existing pipeline is incremental (new step after notch filter)
- Main challenge: tuning reconstruction error threshold for anomaly detection
